File: fixtures/elixir_output/put_basic_auth_json_data.ex
request = %HTTPoison.Request{
  method: :put,
  url: "http://localhost:5984/test/_security",
  options: [hackney: [basic_auth: {~s|admin|, ~s|123|}]],
  headers: [],
  params: [],
  body: ~s|{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}|
}

response = HTTPoison.request(request)

File: test/fixtures/node-request/put_basic_auth_json_data.js
var request = require('request');

var headers = {
    'Content-Type': 'application/x-www-form-urlencoded'
};

var dataString = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}';

var options = {
    url: 'http://localhost:28139/test/_security',
    method: 'PUT',
    headers: headers,
    body: dataString,
    auth: {
        'user': 'admin',
        'pass': '123'
    }
};

function callback(error, response, body) {
    if (!error && response.statusCode == 200) {
        console.log(body);
    }
}

request(options, callback);
